lib/puppet/indirector/catalog/compiler.rb

```diff
@@ -123,6 +121,18 @@ def replace_metadata(resource, metadata)
     when "use_when_creating"
       raise Puppet::Error, "source_permissions => use_when_creating cannot be set when creating static catalogs"
     end
+
+    # The static file content server doesn't know how to expand
+    # mountpoints, so we need to do that ourselves from the actual
+    # system math of the source file. This does that, while preserving
+    # any user-specified server or port.
+    source_path = Pathname.new(metadata.full_path)
+    environment_path = Pathname.new(environment_path)
+    path = source_path.relative_path_from(environment_path).to_s
+    source_as_uri = URI.parse(CGI.escape(metadata.source))
```

Comment:
We should probably wrap it in a try and then print a warning about being unable to inline the specific resource, rather than let the exception fail the whole compile